includes/core-changes/interop/5.0/casting-rcw-to-inspectable-interface-throws-exception.md

```diff
@@ -0,0 +1,53 @@
+### Casting RCW to an `InterfaceIsIInspectable` interface throws PlatformNotSupportedException
+
+Casting a runtime callable wrapper (RCW) to an interface marked as <xref:System.Runtime.InteropServices.ComInterfaceType.InterfaceIsIInspectable> now throws a <xref:System.PlatformNotSupportedException>.
+
+#### Version introduced
+
+5.0 RC2
+
+#### Change description
+
+In previous .NET versions, you can succesfully case an RCW to an interface marked as <xref:System.Runtime.InteropServices.ComInterfaceType.InterfaceIsIInspectable>. However, you might get access violations when you execute methods on the interface.
+
+In .NET 5.0 and later versions, casting an RCW to an interface marked as <xref:System.Runtime.InteropServices.ComInterfaceType.InterfaceIsIInspectable> throws a <xref:System.PlatformNotSupportedException> at cast time. This change is a follow up to the [removal of WinRT support from .NET](../../../../docs/core/compatibility/interop.md#built-in-support-for-winrt-is-removed-from-net).
+
+#### Reason for change
+
+The support for <xref:System.Runtime.InteropServices.ComInterfaceType.InterfaceIsIInspectable> was [removed in a previous .NET 5.0 preview](../../../../docs/core/compatibility/interop.md#built-in-support-for-winrt-is-removed-from-net). However, casting to an <xref:System.Runtime.InteropServices.ComInterfaceType.InterfaceIsIInspectable> interface was accidentally overlooked. Since the underlying support in the runtime no longer exists, throwing a <xref:System.PlatformNotSupportedException> enables a graceful failure path. Throwing an exception also makes it more discoverable that this feature is no longer supported.
+
+#### Recommended action
+
+- If you can define the interface in a Windows runtime metadata (WinMD) file, use the C#/WinRT tool instead.
+
+- Otherwise, mark the interface as <xref:System.Runtime.InteropServices.ComInterfaceType.InterfaceIsIUnknown> instead of <xref:System.Runtime.InteropServices.ComInterfaceType.InterfaceIsIInspectable>, and add three dummy entries to the start of the interface for the <xref:System.Runtime.InteropServices.ComInterfaceType.InterfaceIsIInspectable> methods. The following code snippet shows an example.
+
+  ```csharp
+  [InterfaceType(ComInterfaceType.InterfaceIsIUnknown)]
+  interface IMine
+  {
+      // Do not call these three methods.
+      // They're exclusively to fill in the slots in the vtable.
+      void GetIIdsSlot();
+      void GetRuntimeClassNameSlot();
+      void GetTrustLevelSlot();
+
+      ...
```

Comment:
I removed the `Frob()` method from your example. Is that okay?

Reply:
We should keep `Frob()` since it would have been the first member in the original definition of the interface. Otherwise, something like the following would get the same point across:
```suggestion
      // The original members of the IMine interface go here.
      ...
```